public/app/features/alerting/unified/api/alertingApi.ts

```diff
@@ -82,6 +82,16 @@ export const backendSrvBaseQuery =
 export const alertingApi = createApi({
   reducerPath: 'alertingApi',
   baseQuery: backendSrvBaseQuery(),
+  // The `BasyQueryFn`` passes all args to `getBackendSrv().fetch()` and that includes configuration options for controlling
+  // when to show a "toast".
+  //
+  // By passing "requestOptions" such as "successMessage" etc those also get included in the cache key because
+  // those args are eventually passed in to the baseQueryFn where the cache key gets computed.
+  //
+  // @TODO
+  // Ideally we wouldn't pass any args in to the endpoint at all and toast message behaviour should be controlled
+  // in the hooks or components that consume the RTKQ endpoints.
+  serializeQueryArgs: (args) => defaultSerializeQueryArgs(omit(args, 'queryArgs.requestOptions')),
```

Comment:
`requestOptions` seems confusing here, perhaps we should rename it to `notificationOptions` or similar?

Reply:
No they're not which is why this code is confusing :) I've update the PR to use `notificationOptions` instead which only contains error and success message options.